### PR TITLE
Fetaure/review fix

### PR DESCRIPTION
**Pull request by Rajmohanrk**
login and edit changes

### CODE REVIEW

File: Maxis/wwwroot/login/login.component.html [hunk 1]
```diff
@@ -1,38 +1,27 @@
...
                 <input #password="ngModel" required type="password" class="form-control" name="password" [(ngModel)]="user.password"
                        placeholder="Password" />
 
             </div>
-
-
             <div class="form-group">
-
                 <button class="btn waves-effect waves-light"
                         type="submit" [disabled]="form.invalid" style="background-color:#ADFF2F">
```
Comment (kamalakar1981) [inline Maxis/wwwroot/login/login.component.html]: remove inline style

File: Maxis/wwwroot/login/login.component.ts [hunk 2]
```diff
@@ -13,12 +13,12 @@ export class LoginComponent {
 
     user = new User('', '');
     loginInvalid = false;
-    constructor(private autenticationService: AuthenticationService, private router: Router) {
+    constructor(private _autenticationService: AuthenticationService, private router: Router) {
```
Comment (kamalakar1981) [inline Maxis/wwwroot/login/login.component.ts]: use _ for private variable

File: Maxis/wwwroot/user/edit/edit.component.html [hunk 3]
```diff
@@ -1,6 +1,4 @@
-﻿
-
-<nav class="navbar navbar-light" style="background-color:#66CDAA;margin-bottom: 0px;">
+﻿<nav class="navbar navbar-light" style="background-color:#66CDAA;margin-bottom: 0px;">
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.html]: inline style

File: Maxis/wwwroot/user/edit/edit.component.html [hunk 4]
```diff
@@ -1,6 +1,4 @@
...
     <div class="container-fluid">
         <div class="navbar-header">
             <a class="navbar-brand" href="#"><span class="glyphicon glyphicon-cog" style="color:white"> UserDetails</span></a>
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.html]: inline style

File: Maxis/wwwroot/user/edit/edit.component.html [hunk 5]
```diff
@@ -97,7 +90,6 @@
                     </div>
                     <div class="form-group">
                         <label class="control-label col-sm-2" for="email" style="color:red">Status:</label>
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.html]: inline style

File: Maxis/wwwroot/user/edit/edit.component.html [hunk 6]
```diff
@@ -106,27 +98,15 @@
...
-
             </div>
-
         </form>
-
-
     </div>
-
 </div>
 <div class="panel-footer clearfix" style="margin-top:200px; ">
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.html]: inline style

File: Maxis/wwwroot/user/list/list.component.html [hunk 7]
```diff
@@ -20,8 +20,6 @@
     <div class="row header" style="text-align:center;color:green">
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/list/list.component.html]: inline style

File: Maxis/Views/Shared/_Layout.cshtml [hunk 8]
```diff
@@ -10,8 +10,17 @@
     <script src="~/libs/zone.js/dist/zone.js"></script>
     <script src="~/libs/reflect-metadata/Reflect.js"></script>
     <script src="~/libs/systemjs/dist/system.src.js"></script>
+
+
+    <link rel="stylesheet" href="https://www.primefaces.org/primeng/resources/primeng.min.css"></link>
+    <link rel="stylesheet" href="https://www.primefaces.org/primeng/showcase/resources/icons/css/font-awesome.min.css"></link>
+
+    <link rel="stylesheet" href="https://npmcdn.com/primeui/themes/delta/theme.css"></link>
```
Comment (kamalakar1981) [inline Maxis/Views/Shared/_Layout.cshtml]: remove cdn links

File: Maxis/wwwroot/app.module.ts [hunk 9]
```diff
@@ -14,27 +14,35 @@ import { EditService } from './user/edit/shared/edit.service';
...
 import { Router } from '@angular/router';
-
+import { DataTableModule, SharedModule } from 'primeng/primeng';
 
 @NgModule({
     imports: [BrowserModule,
         routing,
         HttpModule,
         FormsModule,
+        routing,
```
Comment (kamalakar1981) [inline Maxis/wwwroot/app.module.ts]: routing is added twice

File: Maxis/wwwroot/login/login.component.ts [hunk 10]
```diff
@@ -2,35 +2,36 @@
...
-
-    user = new User('', '');
+    errorMsg:string = "Username and password does not exist in LDAP";
+    user: any = {};
     loginInvalid = false;
-    constructor(private autenticationService: AuthenticationService, private router: Router) {
+    constructor(private _autenticationService: AuthenticationService,
+                private _router: Router,
+                private _route: ActivatedRoute ) {
+    constructor(private _autenticationService: AuthenticationService, private router: Router) {
```
Comment (kamalakar1981) [inline Maxis/wwwroot/login/login.component.ts]: why there is 2 constructs ?

File: Maxis/wwwroot/login/shared/authentication.service.ts [hunk 11]
```diff
@@ -22,10 +24,17 @@ export class AuthenticationService {
...
+        //let options = new RequestOptions({ headers: headers });
+
+        let headers = new Headers();
+        headers.append('Content-Type', 'application/json');
+     //   headers.append('Authorization', 'Basic ' + localStorage.getItem("authToken"));
+
         let options = new RequestOptions({ headers: headers });
+        
 
         return this.http.post('http://localhost:56026/Login/Details', body, options)
```
Comment (kamalakar1981) [inline Maxis/wwwroot/login/shared/authentication.service.ts]: remove local host and directly use controller name and method

File: Maxis/wwwroot/login/shared/login.guard.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
+﻿import { Injectable } from '@angular/core';
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+
+@Injectable()
+export class LoginGuard implements CanActivate {
+
+    constructor(private router: Router) { }
```
Comment (kamalakar1981) [inline Maxis/wwwroot/login/shared/login.guard.ts]: use _

File: Maxis/wwwroot/login/shared/login.interface.ts [hunk 13]
```diff
@@ -1,5 +1,4 @@
 ﻿export class User {
-    constructor(
-        public email: string,
-        public password: string) { }
+    email: string;
```
Comment (kamalakar1981) [inline Maxis/wwwroot/login/shared/login.interface.ts]: this should be a interface

File: Maxis/wwwroot/logout/logout.component.ts [hunk 14]
```diff
@@ -1 +1,25 @@
...
+import { Router, ActivatedRoute } from '@angular/router';
+@Component({
+    selector: 'logout',
+    templateUrl: 'wwwroot/logout/logout.component.html',
+    styleUrls: ['wwwroot/logout/logout.component.css']
+})
+
+export class LogoutComponent implements OnInit {
+
+
```
Comment (kamalakar1981) [inline Maxis/wwwroot/logout/logout.component.ts]: remove white space

File: Maxis/wwwroot/logout/shared/logout.service.ts [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+import { Http, Response, Headers, RequestOptions } from '@angular/http';
+import { LogoutComponent } from './../../logout/logout.component';
+import 'rxjs/Rx';
+import { Observable } from 'rxjs/observable';
+@Injectable()
+export class LogoutService {
+
+    constructor(
+        private _http: Http) { }
+
```
Comment (kamalakar1981) [inline Maxis/wwwroot/logout/shared/logout.service.ts]: remove white space and align

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 16]
```diff
@@ -1,13 +1,128 @@
...
+
+    ulist: Userlist;
+    private sub: Subscription;
+    displayMessage: { [key: string]: string } = {};
+    private validationMessages: { [key: string]: { [key: string]: string } };
+
+    constructor(private _editservice: EditService,
+        private fb: FormBuilder,
+        private router: Router,
+        private route: ActivatedRoute) {
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: private variables should start with _

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 17]
```diff
@@ -1,13 +1,128 @@
...
+       
 
-    ngOnInit() {
+        // update the data on the form
+        //this.editform.patchvalue({
+        //    userId: this.ulist.userId,
+        //    userName: this.ulist.userName,
+        //    emailId: this.ulist.emailId,
+        //    mobileNo: this.ulist.mobileNo,
+        //    department: this.ulist.department,
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: clean un used code

File: Maxis/wwwroot/user/edit/shared/edit.guard.ts [hunk 18]
```diff
@@ -0,0 +1,35 @@
+﻿import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, CanDeactivate } from '@angular/router';
+
+import { EditComponent } from './../../edit/edit.component';
+
+@Injectable()
+export class UserDetailGuard implements CanActivate {
+
+    constructor(private router: Router) {
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/shared/edit.guard.ts]: _

File: Maxis/wwwroot/user/edit/shared/edit.service.ts [hunk 19]
```diff
@@ -2,11 +2,11 @@
 import { Http, Response, Headers, RequestOptions } from '@angular/http';
 import 'rxjs/Rx';
 import { Observable } from 'rxjs/observable';
-
-
+import { Userlist } from './../../list/shared/list.interface';
 
 @Injectable()
 export class EditService {
+    private baseUrl = 'api/products';
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/shared/edit.service.ts]: where is this referring to?

File: Maxis/Views/Shared/_Layout.cshtml (hunk 8 above)
Comment (Rajmohanrk) [inline Maxis/Views/Shared/_Layout.cshtml]: namratha is taking care of doing it .

File: Maxis/wwwroot/app.module.ts (hunk 9 above)
Comment (Rajmohanrk) [inline Maxis/wwwroot/app.module.ts]: This issue has been resolved.

**Pull request merged**